### PR TITLE
test: Check that progress logger callback is being called

### LINKED ISSUE #277: Test for logger_callback missing
There is no test which verifies `logger_callback` is being called. (`logger_callback` updates the "% sampling complete" information available to clients.)

### PR DESCRIPTION
**Pull request by riddell-stan**
Closes #277

### CODE REVIEW

File: tests/test_logger_callback.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+"""Test sampling from Bernoulli model."""
+import aiohttp
+import pytest
+
+import helpers
+
+program_code = "parameters {real y;} model {y ~ normal(0,1);}"
```
Comment (ahartikainen) [inline tests/test_logger_callback.py:7]: Should we import these simple models from somewhere (e.g `helpers.py`)

Also do we want to use cached versions or always compile new models?

File: tests/test_logger_callback.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+
+@pytest.mark.asyncio
+async def test_logger_callback(api_url: str) -> None:
+    """Test that operation progress has been updated."""
+
+    payload = {"function": "stan::services::sample::hmc_nuts_diag_e_adapt"}
+    operation = await helpers.sample(api_url, program_code, payload)
+    print(operation)
```
Comment (ahartikainen) [inline tests/test_logger_callback.py:16]: is this debug line?

Comment (riddell-stan) [reply tests/test_logger_callback.py:16]: Good catch. I need to remove that.

File: tests/test_logger_callback.py (hunk 1 above)
Comment (riddell-stan) [reply tests/test_logger_callback.py:7]: The cached models get used automatically when they exist. This seems to work well. We know that it compiled successfully (at least once).

Comment (riddell-stan) [reply tests/test_logger_callback.py:7]: Putting the program_code strings for common models in helpers does sound like a good idea. Maybe in a separate PR though.

File: tests/test_logger_callback.py (hunk 2 above)
Comment (riddell-stan) [reply tests/test_logger_callback.py]: Removed.

**Pull request merged**